src/boundaries/edgetracking.jl

```diff
@@ -0,0 +1,304 @@
+* `success::Bool`: indicates whether the edge tracking has been successful or not
+
+Sometimes, the AttractorMapper used in the algorithm may erroneously identify both states
+`u1` and `u2` with the same basin of attraction due to being very close to the basin
+boundary. If this happens, a warning is raised and `EdgeTrackingResults.success = false`.
+"""
+function edgetracking(ds::DynamicalSystem, attractors::Dict;
+    bisect_thresh=1e-6,
+    diverge_thresh=1e-5,
+    u1=collect(values(attractors))[1][1],
+    u2=collect(values(attractors))[2][1],
+    maxiter=100,
+    abstol=0.0,
+    T_transient=0.0,
+    tmax=Inf,
+    Δt=0.01,
+    ϵ_mapper=nothing,
+    show_progress=true,
+    kwargs...)
+    
+    pds = ParallelDynamicalSystem(ds, [u1, u2])
+    mapper = AttractorsViaProximity(ds, attractors, ϵ_mapper; kwargs...)
+    
+    edgetracking(pds, mapper;
+        bisect_thresh, diverge_thresh, maxiter, abstol, T_transient, Δt, tmax, show_progress)
+end;
+
+"""
+    edgetracking(pds::ParallelDynamicalSystem, mapper::AttractorMapper; kwargs...)
+Low-level function for running the edge tracking algorithm, see [`edgetracking`](@ref)
+for a description, keyword arguments and output type.
+
+`pds` is a `ParallelDynamicalSystem` with two states. The `mapper` must be an
+`AttractorMapper` of subtype `AttractorsViaProximity` or `AttractorsViaRecurrences`.
+"""
+function edgetracking(pds::ParallelDynamicalSystem, mapper::AttractorMapper;
+    bisect_thresh=1e-6,
+    diverge_thresh=1e-5,
+    maxiter=100,
+    abstol=0.0,
+    T_transient=0.0,
+    Δt=0.01,
+    tmax=Inf,
+    show_progress=true)
+    
+    if bisect_thresh >= diverge_thresh
+        error("diverge_thresh must be larger than bisect_thresh.")
+    end
+
+    # initial bisection
+    u1, u2, success = bisect_to_edge(pds, mapper; bisect_thresh)
+    if ~success
+        return EdgeTrackingResults(nothing)
+    end
+    edgestate = (u1 + u2)/2
+    track1, track2 = [u1], [u2]
+    time, bisect_idx = Float64[], Int[]
+    progress = ProgressMeter.Progress(maxiter; desc = "Running edge tracking algorithm",
+        enabled = show_progress)
+    
+    # edge track iteration loop
+    displacement, counter, T = Inf, 1, 0.0
+    while (displacement > abstol) && (maxiter > counter)
+        t = 0
+        set_state!(pds, u1, 1)
+        set_state!(pds, u2, 2)
+        distance = diffnorm(pds)
+        # forward integration loop
+        while (distance < diverge_thresh) && (t < tmax)
+            step!(pds, Δt)
+            distance = diffnorm(pds)
+            t += Δt
+            T += Δt
+            if T >= T_transient
+                push!(track1, current_state(pds, 1))
+                push!(track2, current_state(pds, 2))
+                push!(time, T)
+            end
+        end
+        # re-bisect
+        u1, u2, success = bisect_to_edge(pds, mapper; bisect_thresh)
+        if ~success
+            track1 = StateSpaceSet(reduce(hcat, track1)')
```

Comment:
Why are you doing `reduce(hcat, track1)'` instead of just giving in `track1`...? `StateSpaceSet` accepts a vector of vectors as an input directly, and that's what `track1` is.